RTE Block Configuration and usage (#5562)
umbraco/UmbracoDocs: 12/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-blocks.md

# Blocks in Rich Text Editor

It is possible to insert Blocks into the markup of the Rich Text Editor (RTE). Once you've defined the Block Type as part of the RTE Data Type and enabled the Blocks Toolbar Option. Then Blocks can be created directly inside the Rich Text Editor.

## Configure Blocks

The Block List property editor is configured in the same way as any standard property editor, via the _Data Types_ admin interface.

To set up your Block List Editor property, create a new _Data Type_ and select **Block List** from the list of available property editors.

Then you will see the configuration options for a Block List as shown below.

![Rich Text Editor - Data Type Block Fields](./images/rte-data-type-block-fields.jpg)

The Data Type editor allows you to configure the following properties:

* **Available Blocks** - Here you will define the Block Types to be available for use in the property. Read more on how to set up Block Types below.
* **Blocks Live editing mode** - Enabling this will make editing of a Block happening directly to the Rich Text Editor, making changes appear as you type.

## Setup Block Types

> Block Types are **Element Types** which need to be created before you can start configuring them as Block Types. This can be done either directly from the property editor setup process, or you can set them up beforehand and add them to the Rich Text Editor after.

Reviewer: ```suggestion
Block Types are **Element Types** which need to be created before you can start configuring them as Block Types. This can be done either directly from the property editor setup process, or you can set them up beforehand. If they are set beforehand then they need to be added to the Rich Text Editor afterward.
```

> Once you have added an Element Type as a Block Type on your Rich Text Editor Data Type you will have the option to configure it further.

Reviewer: ```suggestion
Once you added an Element Type as a Block Type on your Rich Text Editor Data Type you will have options to configure it further.
```

> ![Rich Text Editor - Data Type Block Configuration](./images/rte-data-type-block-type-editor.jpg)

Reviewer 2: Seems like the file reference is broken:
```suggestion
![Rich Text Editor - Data Type Block Configuration](./images/rte-data-type-block-type-editor.jpeg)
```

Reviewer 3: Images should probably just be in changed to `jpg` instead to be consistent with the other screenshots.

Reviewer 4: I agree with Bjarne, lets keep it consistent.

* **Overlay editor size** - Set the size for the Content editor overlay for editing this block.

### Data Models

It is possible to use two separate Element Types for your Block Types. Its required to have one for Content and optional to add one for Settings.

* **Content model** - This presents the Element Type used as model for the content section of this Block. This cannot be changed, but you can open the Element Type to perform edits or view the properties available. Useful when writing your Label.
* **Settings model** - Add a Settings section to your Block based on a given Element Type. When picked you can open the Element Type or choose to remove the settings section again.

### Catalogue appearance

These properties refer to how the Block is presented in the Block catalogue, when editors choose which Blocks to use for their content.

* **Background color** - Define a background color to be displayed beneath the icon or thumbnail. Eg. `#424242`.
* **Icon color** - Change the color of the Element Type icon. Eg. `#242424`.

> * **Thumbnail** - Pick an image or SVG file to replace the icon of this Block in the catalogue.

Reviewer: ```suggestion
* **Thumbnail** - Pick an image or Scalable Vector Graphics (SVG) file to replace the icon of this Block in the catalogue.
```
